src/clib/pio_darray_int.c

```diff
@@ -1521,8 +1521,7 @@ int flush_output_buffer(file_desc_t *file, bool force, PIO_Offset addsize)
 
     /* Find out the buffer usage. If I check the turn code, some tests fail. ??? */
     if ((ierr = ncmpi_inq_buffer_usage(file->fh, &usage)))
-        return ierr;
-    /*return pio_err(ios, file, PIO_EBADID, __FILE__, __LINE__);*/
+	return ierr;
```

Comment:
That makes a lot of sense!

It was commented out, as indicated by the comment, because when I put it in, some tests failed. I had not yet figured out why, but it could be the same bug you found.

Please check it, because, if true, it means that we already have a test that detects that problem. ;-)